src/AdminConsole/Services/ApplicationService.cs

```diff
@@ -64,9 +68,14 @@ public async Task CancelDeletionForApplicationAsync(string applicationId)
     public async Task DeleteAsync(string applicationId)
     {
         await using var db = await _dbContextFactory.CreateDbContextAsync();
-        var application = new Application { Id = applicationId };
+        var application = await db.Applications.SingleOrDefaultAsync(x => x.Id == applicationId);
+        if (application == null) return;
         db.Applications.Remove(application);
-        await db.SaveChangesAsync();
+        var rows = await db.SaveChangesAsync();
+        if (rows > 0 && application.BillingSubscriptionItemId != null)
+        {
+            await _billingService.OnPostApplicationDeletedAsync(application.BillingSubscriptionItemId);
```

Comment:
If you have two Pro apps, do they have the same BillingSubscriptionItemId?

What happens to billing for the second app when we delete BillingSubsciptionItemId?

Reply:
It will query again the organization to check if any apps use the subscription item id before removing it in stripe.

`
public async Task OnPostApplicationDeletedAsync(string subscriptionItemId)
    {
        var organization = await _dataService.GetOrganizationWithDataAsync();
        
        // check if other apps use it
        var isSubscriptionItemInUse = organization.Applications.Any(x => x.BillingSubscriptionItemId == subscriptionItemId);
        if (!isSubscriptionItemInUse)
        {
            if (organization.Applications.Any())
            {

                // If we have applications, then we can delete the subscription item,
                // as Stripe requires at least one active subscription item in a subscription.
                var service = new SubscriptionItemService();
                await service.DeleteAsync(subscriptionItemId);
            }
            else
            {
                // no other are in the org, delete the subscription
                var subscriptionItemService = new SubscriptionItemService();
                var subscriptionItem = await subscriptionItemService.GetAsync(subscriptionItemId);
                var subscriptionService = new SubscriptionService();
                var cancelOptions = new SubscriptionCancelOptions();
                cancelOptions.Prorate = false;
                cancelOptions.InvoiceNow = true;
                await subscriptionService.CancelAsync(subscriptionItem.Subscription);
            }
        }
    }
`